src/yaml.ts

```diff
@@ -0,0 +1,46 @@
+import fs from "fs";
+
+import { parse, ScalarTag } from "yaml";
+
+// Use a custom tag for floats in exponential notation, to make the +/- mandatory
+// This fixes commit SHAs consisting only of numbers with a single "e" in them
+export const floatExp: ScalarTag = {
+  identify: (value) => typeof value === "number",
+  default: true,
+  tag: "tag:yaml.org,2002:float",
+  format: "EXP",
+  test: /^[-+]?(?:[0-9][0-9_]*)?(?:\.[0-9_]*)?[eE][-+][0-9]+$/, // Change is here, making the [-+] mandatory
+  resolve: (str: string) => parseFloat(str.replace(/_/g, "")),
+  stringify(node) {
+    const num = Number(node.value);
+    if (isFinite(num)) {
+      return num.toExponential();
+    }
+
+    return isNaN(num) ? ".nan" : num < 0 ? "-.inf" : ".inf";
+  },
+};
+
+export function parseYaml<T>(src: string): T {
+  return parse(src, {
+    version: "1.1", // CodeQL CLI uses YAML 1.1
```

Comment:
Just to confirm, specifying `version: "1.1"` here isn't enough to make the library use the regex from the YAML 1.1 specification? What exactly does this version value control?

Reply:
Unfortunately not, the library [does not use](https://github.com/eemeli/yaml/blob/07063bc865b862c2dc375630e27078b2886d4668/src/schema/yaml-1.1/float.ts#L24) the exact regex from the YAML 1.1 specification, so we need to override it. There do seem to be some changes between [the regex used for YAML 1.2 in the library](https://github.com/eemeli/yaml/blob/07063bc865b862c2dc375630e27078b2886d4668/src/schema/core/float.ts#L24), so there are some differences in parsing.